src/diploma.c

```diff
@@ -0,0 +1,254 @@
+#include "global.h"
+#include "bg.h"
+#include "constants/fanfares.h"
+#include "dynamic_placeholder_text_util.h"
+#include "gba/macro.h"
+#include "gpu_regs.h"
+#include "gpu_regs.h"
+#include "main.h"
+#include "malloc.h"
+#include "menu.h"
+#include "new_menu_helpers.h"
+#include "overworld.h"
+#include "palette.h"
+#include "pokedex.h"
+#include "scanline_effect.h"
+#include "sound.h"
+#include "sprite.h"
+#include "task.h"
+#include "text.h"
+#include "window.h"
+#include "diploma.h"
+
+static void DiplomaBgInit(void);
+static void DiplomaPrintText(void);
+static u8 DiplomaLoadGfx(void);
+static void DiplomaVblankHandler(void);
+
+static void CB2_DiplomaInit(void);
+
+static void Task_WaitForExit(u8);
+static void Task_DiplomaInit(u8);
+static void Task_DiplomaReturnToOverworld(u8);
+
+extern const struct BgTemplate gUnknown_8415A08[2];
+extern const struct WindowTemplate gUnknown_8415A10[];
```

Comment:
Usually we decompile data as well. You probably will want to decompile the 2 templates, (with change in ld_script.txt and some file in data, because these data are directly INCBINed from baserom.gba there) and document them with names starting with lowercase `s` because of static.